collect_app/src/main/java/org/odk/collect/android/widgets/items/SelectOneFromMapDialogFragment.kt

```diff
@@ -0,0 +1,137 @@
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        DaggerUtils.getComponent(context).inject(this)
+
+        childFragmentManager.fragmentFactory = FragmentFactoryBuilder()
+            .forClass(SelectionMapFragment::class.java) {
+                val formIndex = requireArguments().getSerializable(ARG_FORM_INDEX) as FormIndex
+                val prompt = formEntryViewModel.getQuestionPrompt(formIndex)
+                SelectionMapFragment(
+                    SelectChoicesMapData(prompt),
+                    skipSummary = true,
+                    showNewItemButton = false
+                )
+            }
+            .build()
+
+        childFragmentManager.setFragmentResultListener(REQUEST_SELECT_ITEM, this, this)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        val binding = SelectOneFromMapDialogLayoutBinding.inflate(inflater)
+        return binding.root
+    }
+
+    override fun getToolbar(): Toolbar? {
+        return null
+    }
+
+    override fun onBackPressed() {
+        dismiss()
+    }
+
+    override fun onCloseClicked() {
+        // No toolbar so not relevant
+    }
+
+    override fun onFragmentResult(requestKey: String, result: Bundle) {
+        val selectedIndex = result.getLong(SelectionMapFragment.RESULT_SELECTED_ITEM).toInt()
+        val formIndex = requireArguments().getSerializable(ARG_FORM_INDEX) as FormIndex
+        val prompt = formEntryViewModel.getQuestionPrompt(formIndex)
+        val selectedChoice = prompt.selectChoices[selectedIndex]
+        formEntryViewModel.answerQuestion(formIndex, SelectOneData(selectedChoice.selection()))
+        dismiss()
+    }
+
+    companion object {
+        const val ARG_FORM_INDEX = "form_index"
+    }
+}
+
+internal class SelectChoicesMapData(prompt: FormEntryPrompt) : SelectionMapData {
+
+    private val mapTitle = MutableLiveData(prompt.longText)
+    private val itemCount = MutableLiveData<Int>()
+    private val items = MutableNonNullLiveData(emptyList<MappableSelectItem>())
+
+    init {
+        val selectChoices = prompt.selectChoices
+        itemCount.value = selectChoices.size
+        items.value = selectChoices.foldIndexed(emptyList()) { index, list, selectChoice ->
+            val geometry = selectChoice.getChild("geometry")
+
+            if (geometry != null) {
+                val latitude = geometry.split(" ")[0].toDouble()
+                val longitude = geometry.split(" ")[1].toDouble()
+
+                list + MappableSelectItem.WithInfo(
+                    index.toLong(),
+                    latitude,
+                    longitude,
+                    R.drawable.ic_map_marker_24dp,
+                    R.drawable.ic_map_marker_48dp,
+                    selectChoice.labelInnerText,
```

Comment:
@lognaturel @grzesiek2010 should this use `FormEntryPrompt#getSelectChoiceText` instead? Also, in #4948 we described displaying the `name` if `label` was absent/blank - it doesn't look like `getSelectChoiceText` does this, so I'm not sure why we're adding that flexibility here and not to normal selects?

Reply:
I don't know I vaguely remember something that there were cases where `attachChoice` which is a part of `getSelectChoiceText` was needed but now I can't say anything for sure.

Reply:
`labelInnerText` won't work for localized forms. [Here is an example](https://docs.google.com/spreadsheets/d/1j3COXsYsgA4tX4d922ho19Qm0jn9Xr7NnjxaTc6kJV4/edit?usp=sharing). With that form none of the points are showing up on the map. Are forms without labels maybe filtered out currently?

I had described the extra flexibility because the geojson spec only talks about an optional `id` and does not define any way of including a label. We've adopted the ["simple style spec"](https://github.com/mapbox/simplestyle-spec/tree/master/1.1.0) `title` standard but that's not widely used.

We could require the label initially and relax that later if we hear that it's important. If we do that, we might want to do it for all selects.

Reply:
That sounds good. Right now I'll fix this up and edit out the fallback from `label` to `id` in #4948.